src/structure/structure.ts

```diff
@@ -33,22 +33,27 @@ function groupByStructure(n_structures: number, environments?: Environment[]): E
  * issue with `s` if any, or the empty string if `s` looks like a valid
  * structure.
  */
-function checkStructure(s: any): string {
+function checkStructure(s: JsObject): string {
```

Comment:
Would it be possible to move this function to `dataset.ts` and employ it in dataset.checkStructures? The two functions contain a lot of the same code.

Reply:
Good idea!